### DISCUSSION TITLE
Works on windows version but not linux

### DISCUSSION THREAD

**Discussion by campones** [streamlink/streamlink#5134] [Q&A] [answered]
### Description

Hello

I m facing a curious issue today. 

While trying to use streamlink (or ffmpeg) with an hls stream, the only platform it works is my windows computer. I have tried several linux computer, including the one beside me (sharing the same residential ip address then) 
both my windows computer and home linux have the last streamlink version. 
On all linux I get 403 forbidden, while it just works fine on windows.  Also you can input the stream url into a web player to the world and it works!? (I was confirmed it works from Quebec so it's not geo-blocked), so I m really wondering what could cause this, and what could be the work around, some cmd line options? 

Using Linux 

streamlink "hlsvariant://https://ncdn-live-bfm.pfd.sfr.net/shls/LIVE$BFM_TV/index.m3u8?start=LIVE&end=END" best

[cli][info] Found matching plugin hls for URL hlsvariant://https://ncdn-live-bfm.pfd.sfr.net/shls/LIVE/index.m3u8?start=LIVE&end=END
error: Unable to open URL: https://ncdn-live-bfm.pfd.sfr.net/shls/LIVE/index.m3u8?start=LIVE&end=END (403 Client Error: Forbidden for url: https://cbv4-cdn-edge-live03.pfd.sfr.net/bfm-ncdn-live-pal1.pfd.sfr.net/sid=0896p4mkvvfb7m4q6bo0/shls/LIVE/index.m3u8?start=LIVE&end=END)

Using windows10

C:\Users\campones>streamlink "hlsvariant://https://ncdn-live-bfm.pfd.sfr.net/shls/LIVE$BFM_TV/index.m3u8?start=LIVE&end=END" best
[cli][info] Found matching plugin hls for URL hlsvariant://https://ncdn-live-bfm.pfd.sfr.net/shls/LIVE$BFM_TV/index.m3u8?start=LIVE&end=END
[cli][info] Available streams: 360p_alt (worst), 360p, 576p, 720p_alt, 720p, 1080p (best)
[cli][info] Opening stream: 1080p (hls)
[cli][info] Starting player: "C:\Program Files (x86)\VideoLAN\VLC\vlc.exe"

thanks for any hint about this

**Comment by bastimeyer** [comment, marked as answer]
> `streamlink "hlsvariant://https://ncdn-live-bfm.pfd.sfr.net/shls/LIVE$BFM_TV/index.m3u8?start=LIVE&end=END" best`
> `error: Unable to open URL: https://ncdn-live-bfm.pfd.sfr.net/shls/LIVE/index.m3u8?start=LIVE&end=END`

The input URL contains a `$` character, which gets interpreted differently on a POSIX compliant shell, namely as the prefix of a variable. On Windows' batch CMD shell, variables are surrounded by `%` characters.

You'll either have to escape the `$` character, or use a single-quote string, to make the shell ignore special characters like this. Learn how shell escaping works.
https://wiki.bash-hackers.org/syntax/quoting

```
$ streamlink -l debug 'https://ncdn-live-bfm.pfd.sfr.net/shls/LIVE$BFM_TV/index.m3u8?start=LIVE&end=END'
[cli][debug] OS:         Linux-6.1.8-1-git-x86_64-with-glibc2.36
[cli][debug] Python:     3.11.1
[cli][debug] Streamlink: 5.2.1+6.g2e17b514
[cli][debug] Dependencies:
[cli][debug]  certifi: 2022.12.7
[cli][debug]  isodate: 0.6.1
[cli][debug]  lxml: 4.9.2
[cli][debug]  pycountry: 22.3.5
[cli][debug]  pycryptodome: 3.16.0
[cli][debug]  PySocks: 1.7.1
[cli][debug]  requests: 2.28.2
[cli][debug]  urllib3: 1.26.14
[cli][debug]  websocket-client: 1.4.2
[cli][debug]  importlib-metadata: 6.0.0
[cli][debug] Arguments:
[cli][debug]  url=https://ncdn-live-bfm.pfd.sfr.net/shls/LIVE$BFM_TV/index.m3u8?start=LIVE&end=END
[cli][debug]  --loglevel=debug
[cli][debug]  --player=mpv
[cli][info] Found matching plugin hls for URL https://ncdn-live-bfm.pfd.sfr.net/shls/LIVE$BFM_TV/index.m3u8?start=LIVE&end=END
[plugins.hls][debug] URL=https://ncdn-live-bfm.pfd.sfr.net/shls/LIVE$BFM_TV/index.m3u8?start=LIVE&end=END; params={}
[utils.l10n][debug] Language code: en_US
Available streams: 360p_alt (worst), 360p, 576p, 720p_alt, 720p, 1080p (best)
```

**Reply by campones** [reply]
Great answer.
Well I mostly use " instead of ' 
That's why it wasn't working, thanks!
